src/impl_clone.rs

```diff
@@ -5,16 +5,18 @@
 // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
+use std::ptr::NonNull;
+
 use crate::imp_prelude::*;
 use crate::RawDataClone;
 
 impl<S: RawDataClone, D: Clone> Clone for ArrayBase<S, D> {
     fn clone(&self) -> ArrayBase<S, D> {
         unsafe {
-            let (data, ptr) = self.data.clone_with_ptr(self.ptr);
+            let (data, ptr) = self.data.clone_with_ptr(self.ptr.as_ptr());
```

Comment:
Yes why not change the clone_with_ptr signature, so that it uses NonNull<T> all the way through?